weasyl/character.py

```diff
@@ -186,60 +190,115 @@ def is_hidden(charid):
     return bool(results and results[0][0])
 
 
-def select_view(userid, charid, rating, ignore=True, anyway=None):
-    query = define.execute("""
+def _select_character_and_check(userid, charid, rating=None, ignore=True, anyway=False, increment_views=True):
+    """Selects a character, after checking if the user is authorized, etc.
+
+    Args:
+        userid (int): Currently authenticating user ID.
+        charid (int): Character ID to fetch.
+        rating (int): Maximum rating to display.
+        ignore (bool): If should respect ignored or blocked tags.
+        anyway (bool): If should ignore checks and display anyway.
+        increment_views (bool): If should increment the number of views on the submission.
+
+    Returns:
+        A character and all needed data.
+    """
+
+    query = define.engine.execute("""
         SELECT
             ch.userid, pr.username, ch.unixtime, ch.char_name, ch.age, ch.gender, ch.height, ch.weight, ch.species,
             ch.content, ch.rating, ch.settings, ch.page_views, pr.config
         FROM character ch
             INNER JOIN profile pr USING (userid)
-        WHERE ch.charid = %i
-    """, [charid], options=["single", "list"])
+        WHERE ch.charid = %(charid)s
+    """, charid=charid).fetchone()
 
-    if query and userid in staff.MODS and anyway == "true":
+    if query and userid in staff.MODS and anyway:
         pass
-    elif not query or "h" in query[11]:
-        raise WeasylError("characterRecordMissing")
-    elif query[10] > rating and ((userid != query[0] and userid not in staff.MODS) or define.is_sfw_mode()):
-        raise WeasylError("RatingExceeded")
-    elif "f" in query[11] and not frienduser.check(userid, query[0]):
-        raise WeasylError("FriendsOnly")
-    elif ignore and ignoreuser.check(userid, query[0]):
-        raise WeasylError("UserIgnored")
+    elif not query or 'h' in query.settings:
+        raise WeasylError('characterRecordMissing')
+    elif query.rating > rating and ((userid != query.userid and userid not in staff.MODS) or define.is_sfw_mode()):
```

Comment:
The idea appears to be that mods and the owner should be able to see it regardless of rating unless SFW mode is also on (and that unless `anyway` is `True`). Since it matches the original behaviour, I’d just leave it for now.